### PR TITLE
Fix cmip6 testmods

### PR DESCRIPTION
**Pull request by fischer-ncar**
One the tests was using the cmip6 testmod that shoudn't have been.  And another test should have been using the cmip6 testmod. 

User interface changes?: [ No/Yes ]
[ If yes, describe what changed, and steps taken to ensure backward compatibilty ]

Fixes: [Github issue #s] And brief description of each issue.

Testing:
  unit tests:
  system tests:
  manual testing:

### CODE REVIEW

File: cime_config/testlist_allactive.xml [hunk 1]
```diff
@@ -375,7 +375,7 @@
       <option name="wallclock">  00:30 </option>
     </options>
   </test>
-  <test name="SMS_D_Ld1" grid="f09_g17" compset="B1850" testmods="allactive/cmip6">
```
Comment (billsacks) [inline cime_config/testlist_allactive.xml]: Please revert this change: this is a new test that I added explicitly to test the cmip6 testmod in a B1850 configuration. Was it causing problems?

**Comment by fischer-ncar** [comment]
@billsacks, I revert the test SMS_D_Ld1.f09_g17.B1850 back to cmip6, and I'll not it as an expected failures.
